### DISCUSSION TITLE
libmesh hashing.h

### DISCUSSION THREAD

**Discussion by heinono1** [idaholab/moose#21310] [Q&A Getting Started] [answered]
Hi. I am bulding libmesh from a new dist using update_and_rebuild_libmesh.sh. The build crashes at ~/contrib/fparser_ad.cc: fatal error: libmesh/hashing.h: No such file or directory #include<libmesh/hashing.h>
The hashing.h file does exist in ./libmesh/inlcude/utils and ./libmesh/include/libmeh (reckoned from the moose directory). It is not clear to me where the build is looking for hashing.h and/or how I should set paths to fix this.

**Comment by GiudGiud** [comment]
Hello

Could you please attach the config.log in the scripts folder?
Also there is a diagnostics script in moose/scripts you may run to understand the environment

Guillaume

**Reply by heinono1** [reply]
Hi Guillaume. config.log is attached.
[config.log](https://github.com/idaholab/moose/files/8911447/config.log)

**Reply by GiudGiud** [reply]
hello

You are partially using the intel oneapi compilers.
We dont support intel compilers anymore as they had a bug, we dropped support for them and we havent come back to trying them.

You also have some gcc compilers active there. Please just unload the modules / remove from the path the intel compilers and stick with GNU compilers and openmpi or similar. 

Guillaume

**Comment by heinono1** [comment]
OK, thanks. Will do.

-Olle

On Wed, Jun 15, 2022 at 12:37 PM Guillaume Giudicelli <
***@***.***> wrote:

> hello
>
> You are partially using the intel oneapi compilers.
> We dont support intel compilers anymore as they had a bug, we dropped
> support for them and we havent come back to trying them.
>
> You also have some gcc compilers active there. Please just unload the
> modules / remove from the path the intel compilers and stick with GNU
> compilers and openmpi or similar.
>
> Guillaume
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/idaholab/moose/discussions/21310#discussioncomment-2958010>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AEKZEF7OSI5BNKWLZ3ZFCXLVPIIELANCNFSM5Y34ZAQQ>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
Olle Heinonen
***@***.***

**Comment by heinono1** [comment]
It does not fix the problem. Still the same error re. hashing.h

On Wed, Jun 15, 2022 at 12:37 PM Guillaume Giudicelli <
***@***.***> wrote:

> hello
>
> You are partially using the intel oneapi compilers.
> We dont support intel compilers anymore as they had a bug, we dropped
> support for them and we havent come back to trying them.
>
> You also have some gcc compilers active there. Please just unload the
> modules / remove from the path the intel compilers and stick with GNU
> compilers and openmpi or similar.
>
> Guillaume
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/idaholab/moose/discussions/21310#discussioncomment-2958010>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AEKZEF7OSI5BNKWLZ3ZFCXLVPIIELANCNFSM5Y34ZAQQ>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
Olle Heinonen
***@***.***

**Reply by GiudGiud** [reply]
what's the config log now

**Comment by heinono1** [comment]
Attached. There s a whole slew of header files that codes in ./contrib/fparser (fparser_ad.cc, as well as header files here looking for header files) cannot find.
[config.log](https://github.com/idaholab/moose/files/8912125/config.log)

**Comment by heinono1** [comment]
It looks like the problem originates in
./libmesh/contrib/fparser/fparser_ad.cc - it looks for some headerfile that
it cannot find, and that headerfile in turn is looking for other header
files in the same directory
#include <libmesh/some_header_file.h>

On Wed, Jun 15, 2022 at 1:21 PM Guillaume Giudicelli <
***@***.***> wrote:

> what's the config log now
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/idaholab/moose/discussions/21310#discussioncomment-2958276>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AEKZEF33NSCR7GDQGT3DMILVPINJ3ANCNFSM5Y34ZAQQ>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
Olle Heinonen
***@***.***

**Reply by GiudGiud** [reply]
seems like the same problem @aruybalid was hitting

**Comment by heinono1** [comment]
The problem seems to be a few programs:
./libmesh/contrib/fparser/fparser_ad.cc and ./libmesh/scr/apps/fparser.C
where the paths to header files, which all involve "libmesh/foo.h" or
<libmesh.foo.h>, fail. I am doing the stupid thing to fix this: edit
fparser.C and fparser_ad.cc to look for the header files in the same
directory as the cc or C file - and also edit header files that look for
header files. I copied the relevant header files (which live in ./libmesh I
think) to those two directories. This is obviously a coarse hack but I was
not going to go in end spend more time trying to figure out why the paths
got screwed up.

-Olle

On Wed, Jun 15, 2022 at 6:50 PM Guillaume Giudicelli <
***@***.***> wrote:

> seems like the same problem @aruybalid <https://github.com/aruybalid> was
> hitting
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/idaholab/moose/discussions/21310#discussioncomment-2959857>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AEKZEF3N474RZMA7OXRYXCTVPJT37ANCNFSM5Y34ZAQQ>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
Olle Heinonen
***@***.***

**Comment by heinono1** [comment, marked as answer]
So I just pulled a completely fresh dist, made sure I had loaded only GNU
compilers and compatible openmpi, cmake, have python linked to python3.
Now petsc and libmesh both built - no problem (!!). But when I build moose
(cd ./test && make -j 4) I get the message right away "make: python-config:
Command not found". Any ideas???

On Thu, Jun 16, 2022 at 8:31 AM Olle Heinonen ***@***.***>
wrote:

> The problem seems to be a few programs:
> ./libmesh/contrib/fparser/fparser_ad.cc and ./libmesh/scr/apps/fparser.C
> where the paths to header files, which all involve "libmesh/foo.h" or
> <libmesh.foo.h>, fail. I am doing the stupid thing to fix this: edit
> fparser.C and fparser_ad.cc to look for the header files in the same
> directory as the cc or C file - and also edit header files that look for
> header files. I copied the relevant header files (which live in ./libmesh I
> think) to those two directories. This is obviously a coarse hack but I was
> not going to go in end spend more time trying to figure out why the paths
> got screwed up.
>
> -Olle
>
> On Wed, Jun 15, 2022 at 6:50 PM Guillaume Giudicelli <
> ***@***.***> wrote:
>
>> seems like the same problem @aruybalid <https://github.com/aruybalid>
>> was hitting
>>
>> —
>> Reply to this email directly, view it on GitHub
>> <https://github.com/idaholab/moose/discussions/21310#discussioncomment-2959857>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/AEKZEF3N474RZMA7OXRYXCTVPJT37ANCNFSM5Y34ZAQQ>
>> .
>> You are receiving this because you authored the thread.Message ID:
>> ***@***.***>
>>
>
>
> --
> Olle Heinonen
> ***@***.***
>


-- 
Olle Heinonen
***@***.***

**Reply by milljm** [reply]
by linking python to python3, it (MOOSE) expects to find python-config also linked... Basically do this:

```pre
ln -s /path/to/python3-config /some/path/within PATH/python-config
```

**Comment by heinono1** [comment]
Thanks. The problem is that python-config is not installed. I am talking to
our system manager to get that done (I cannot use a conda install for
myself for complicated reasons).

Cheers,
Olle

On Mon, Jun 20, 2022 at 8:53 AM Jason Miller ***@***.***>
wrote:

> by linking python to python3, it (MOOSE) expects to find python-config
> also linked... Basically do this:
>
> ln -s /path/to/python3-config /some/path/within PATH/python-config
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/idaholab/moose/discussions/21310#discussioncomment-2986151>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AEKZEFY6JYCP57CLFRV6PQTVQBZVBANCNFSM5Y34ZAQQ>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
Olle Heinonen
***@***.***

**Comment by heinono1** [comment]
Got links to python3 and python3-config. Compilation works. Linking fails:

The linker shows a problem: it is “skipping incompatible libraries
/ib/libpthread.so when searching for -lpthread”

Same for -lrt, -ldl, -lm, and -lc.


That is usually because of a compatibility issue - trying to link a 64-bit
build with 32-bit library binaries. But that should not be the case. And
the only modules I have loaded are the GNU compiler (64-bit) and associated
libraries. Any thoughts?


-Olle

On Mon, Jun 20, 2022 at 9:21 AM Olle Heinonen ***@***.***>
wrote:

> Thanks. The problem is that python-config is not installed. I am talking
> to our system manager to get that done (I cannot use a conda install for
> myself for complicated reasons).
>
> Cheers,
> Olle
>
> On Mon, Jun 20, 2022 at 8:53 AM Jason Miller ***@***.***>
> wrote:
>
>> by linking python to python3, it (MOOSE) expects to find python-config
>> also linked... Basically do this:
>>
>> ln -s /path/to/python3-config /some/path/within PATH/python-config
>>
>> —
>> Reply to this email directly, view it on GitHub
>> <https://github.com/idaholab/moose/discussions/21310#discussioncomment-2986151>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/AEKZEFY6JYCP57CLFRV6PQTVQBZVBANCNFSM5Y34ZAQQ>
>> .
>> You are receiving this because you authored the thread.Message ID:
>> ***@***.***>
>>
>
>
> --
> Olle Heinonen
> ***@***.***
>


-- 
Olle Heinonen
***@***.***

**Reply by GiudGiud** [reply]
where is this ib/libpthread coming from? Is this your system one?
@milljm does it come with the compilers?

**Comment by GiudGiud** [comment]
Is this fixed?
And if not could we please see the latest results for the diagnostics script

**Comment by GiudGiud** [comment]
weird. Can you add moose/python to your PYTHONPATH
also what python version are you running?

Guillaume


Le jeu. 16 juin 2022 à 16:46, Olle Heinonen ***@***.***> a
écrit :

> So I just pulled a completely fresh dist, made sure I had loaded only GNU
> compilers and compatible openmpi, cmake, have python linked to python3.
> Now petsc and libmesh both built - no problem (!!). But when I build moose
> (cd ./test && make -j 4) I get the message right away "make: python-config:
> Command not found". Any ideas???
>
> On Thu, Jun 16, 2022 at 8:31 AM Olle Heinonen ***@***.***>
> wrote:
>
> > The problem seems to be a few programs:
> > ./libmesh/contrib/fparser/fparser_ad.cc and ./libmesh/scr/apps/fparser.C
> > where the paths to header files, which all involve "libmesh/foo.h" or
> > <libmesh.foo.h>, fail. I am doing the stupid thing to fix this: edit
> > fparser.C and fparser_ad.cc to look for the header files in the same
> > directory as the cc or C file - and also edit header files that look for
> > header files. I copied the relevant header files (which live in
> ./libmesh I
> > think) to those two directories. This is obviously a coarse hack but I
> was
> > not going to go in end spend more time trying to figure out why the paths
> > got screwed up.
> >
> > -Olle
> >
> > On Wed, Jun 15, 2022 at 6:50 PM Guillaume Giudicelli <

**Comment by heinono1** [comment]
Thanks, Guillaume. I've got all sorted out - and running
